keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/Monikered.kt

```diff
@@ -21,6 +21,8 @@ package com.netflix.spinnaker.keel.api
  * A resource that has a moniker-compliant name.
  */
 interface Monikered : ResourceSpec {
+  val name: String
```

Comment:
I think it would be valuable to have a validation function that took the app name, and made sure the name started with the app name. I worry about this making it really easy to have resources that break our spinnaker assumptions (by accident)

Reply:
Although I agree, we don't do that now with the `moniker` property

Reply:
Are there situations where it's valid to be managing a resource that's outside of the application?

Reply:
Hmm, good point that we don't validate now. I'm not sure - I think the resources we support right now need to be part of valid application. TBD on whether that will change in the future.